--- gettext-0.14.1/gettext-tools/src/x-perl.c.orig	2003-12-29 14:54:11 +0300
+++ gettext-0.14.1/gettext-tools/src/x-perl.c	2005-01-04 15:35:24 +0300
@@ -2459,6 +2459,13 @@ x_perl_prelex (message_list_ty *mlp, tok
 	      tp->type = token_type_dereference;
 	      return;
 	    }
+	  else if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
+	    {
+	      /* One of the -X (filetest) functions.  We play safe
+		 and accept all alphabetical characters here.  */
+	      tp->type = token_type_other;
+	      return;
+	    }
 	  phase1_ungetc (c);
 	  tp->type = token_type_other;
 	  prefer_division_over_regexp = false;
