--- gettext-0.11.5/m4/fnmatch.m4.libobjs	2002-02-04 16:32:49 +0300
+++ gettext-0.11.5/m4/fnmatch.m4	2002-10-12 20:40:16 +0400
@@ -44,7 +44,6 @@
     dnl libgettextlib.so.
     AC_DEFINE([fnmatch], [posix_fnmatch],
       [Define to a replacement function name for fnmatch().])
-    LIBOBJS="$LIBOBJS pfnmatch.${ac_objext}"
-    AC_SUBST(LIBOBJS)
+    AC_LIBOBJ([pfnmatch])
   fi
 ])
--- gettext-0.11.5/m4/getline.m4.libobjs	2002-01-16 15:19:34 +0300
+++ gettext-0.11.5/m4/getline.m4	2002-10-12 20:40:48 +0400
@@ -47,7 +47,6 @@
     dnl libgettextlib.so.
     AC_DEFINE([getline], [gnu_getline],
       [Define to a replacement function name for getline().])
-    LIBOBJS="$LIBOBJS getline.${ac_objext}"
-    AC_SUBST(LIBOBJS)
+    AC_LIBOBJ([getline])
   fi
 ])
--- gettext-0.11.5/configure.in.libobjs	2002-10-12 20:32:18 +0400
+++ gettext-0.11.5/configure.in	2002-10-12 20:43:07 +0400
@@ -226,9 +226,9 @@
 
 dnl These ought to go away some day.
 changequote(,)dnl
-LTLIBOBJS=`echo "$LIBOBJS" | sed 's/\.[^.]* /.lo /g;s/\.[^.]*$/.lo/'`
+dnl LTLIBOBJS=`echo "$LIBOBJS" | sed 's/\.[^.]* /.lo /g;s/\.[^.]*$/.lo/'`
 changequote([, ])dnl
-AC_SUBST(LTLIBOBJS)
+dnl AC_SUBST(LTLIBOBJS)
 changequote(,)dnl
 LTALLOCA=`echo "$ALLOCA" | sed 's/\.[^.]* /.lo /g;s/\.[^.]*$/.lo/'`
 changequote([, ])dnl
