--- gettext-0.14.1/gettext-runtime/m4/signed.m4.orig	2002-01-16 15:19:35 +0300
+++ gettext-0.14.1/gettext-runtime/m4/signed.m4	2004-09-13 18:55:12 +0400
@@ -11,7 +11,7 @@
 AC_DEFUN([bh_C_SIGNED],
 [
   AC_CACHE_CHECK([for signed], bh_cv_c_signed,
-   [AC_TRY_COMPILE(, [signed char x;], bh_cv_c_signed=yes, bh_cv_c_signed=no)])
+   [AC_TRY_COMPILE(, [signed char x; x='x';], bh_cv_c_signed=yes, bh_cv_c_signed=no)])
   if test $bh_cv_c_signed = no; then
     AC_DEFINE(signed, ,
               [Define to empty if the C compiler doesn't support this keyword.])
