--- gettext/autogen.sh
+++ gettext/autogen.sh
@@ -385,16 +385,6 @@ if test -n "$GNULIB_TOOL"; then
   for file in config.guess config.sub; do
     $GNULIB_TOOL --copy-file build-aux/$file; chmod a+x build-aux/$file || exit $?
   done
-else
-  for file in config.guess config.sub; do
-    echo "$0: getting $file..."
-    wget -q --timeout=5 -O build-aux/$file.tmp "http://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob_plain;f=build-aux/${file};hb=HEAD" \
-      && mv build-aux/$file.tmp build-aux/$file \
-      && chmod a+x build-aux/$file
-    retval=$?
-    rm -f build-aux/$file.tmp
-    test $retval -eq 0 || exit $retval
-  done
 fi
 
 # Fetch gettext-tools/misc/archive.dir.tar.
