--- gettext-0.14.1/gettext-tools/misc/autopoint.in.orig	2005-01-04 15:50:56 +0300
+++ gettext-0.14.1/gettext-tools/misc/autopoint.in	2005-01-04 16:00:13 +0300
@@ -277,10 +277,23 @@
 # Set variables
 # - cvs_dir         directory containing the temporary repository
 # - work_dir        directory containing the temporary checkout
-cvs_dir=tmpcvs$$
-work_dir=tmpwrk$$
-mkdir "$cvs_dir"
-mkdir "$work_dir"
+cur_dir=`pwd` || exit
+cvs_dir=
+work_dir=
+
+exit_handler()
+{
+	local rc=$?
+	trap - EXIT
+	[ -z "$cvs_dir" ] || rm -rf -- "$cur_dir/$cvs_dir"
+	[ -z "$work_dir" ] || rm -rf -- "$cur_dir/$work_dir"
+	exit $rc
+}
+
+trap exit_handler HUP PIPE INT QUIT TERM EXIT
+
+cvs_dir="$(mktemp -d tmpcvs.XXXXXXXXXX)" || exit
+work_dir="$(mktemp -d tmpwrk.XXXXXXXXXX)" || exit
 CVSROOT="$srcdir/$cvs_dir"
 export CVSROOT
 unset CVS_CLIENT_LOG
@@ -308,8 +321,6 @@
 ("$CVS_EXE" checkout -r"$cvsver" archive > /dev/null) 2>&1 | grep -v '^cvs checkout: Updating'
 find archive -name CVS -type d -print | xargs rm -rf
 if test `find archive -type f -print | wc -l` = 0; then
-  cd ..
-  rm -rf "$cvs_dir" "$work_dir"
   func_fatal_error "infrastructure files for version $ver not found; this is autopoint from GNU $package $version"
 fi
 cd ..
@@ -347,8 +358,7 @@
 # original - too great risk of version mismatch.
 if test -z "$force"; then
   mismatch=
-  mismatchfile="${TMPDIR-/tmp}"/autopoint$$.diff
-  rm -f "$mismatchfile"
+  mismatchfile="`mktemp -t autopoint.diff.XXXXXX`"
   for file in `find "$work_dir/archive" -type f -print | sed -e "s,^$work_dir/archive/,," | LC_ALL=C sort`; do
     func_destfile "$file"
     if test -n "$destfile"; then
@@ -364,7 +374,6 @@
     fi
   done
   if test -n "$mismatch"; then
-    rm -rf "$cvs_dir" "$work_dir"
     func_fatal_error "Some files have been locally modified. Not overwriting them because --force has not been specified. For your convenience, you find the local modifications in the file '$mismatchfile'."
   fi
 fi
@@ -380,6 +389,10 @@
     # Recompute base. It was clobbered by the recursive call.
     base=`echo "$1" | sed -e 's,/[^/]*$,,'`
     test -d "$base" || { echo "Creating directory $base"; mkdir "$base"; }
+    if [ $? -ne 0 ]; then
+      echo "ERROR making directory $base"
+      exit 1
+    fi
   fi
 }
 
@@ -435,5 +448,4 @@
 done
 
 # That's it.
-rm -rf "$cvs_dir" "$work_dir"
 exit 0
