--- gettext/gettext-tools/gnulib-m4/libxml.m4
+++ gettext/gettext-tools/gnulib-m4/libxml.m4
@@ -42,10 +42,10 @@ AC_DEFUN([gl_LIBXML],
       dnl -Wl,--disable-auto-import.
       AC_LIB_LINKFLAGS_BODY([xml2])
       LIBS="$gl_save_LIBS $LIBXML2 $LIBICONV"
-      AC_TRY_LINK([[#include <libxml/xmlversion.h>
+      AC_TRY_LINK([#include <libxml/xmlversion.h>
                     #include <libxml/xmlmemory.h>
-                  ]],
-        [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                  ],
+        [xmlCheckVersion (0); xmlFree ((void *) 0);],
         [gl_cv_libxml=yes
          gl_cv_LIBXML="$LIBXML2 $LIBICONV"
          gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
@@ -53,10 +53,10 @@ AC_DEFUN([gl_LIBXML],
       if test "$gl_cv_libxml" != yes; then
         gl_save_CPPFLAGS="$CPPFLAGS"
         CPPFLAGS="$CPPFLAGS $INCXML2"
-        AC_TRY_LINK([[#include <libxml/xmlversion.h>
+        AC_TRY_LINK([#include <libxml/xmlversion.h>
                       #include <libxml/xmlmemory.h>
-                    ]],
-          [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                    ],
+          [xmlCheckVersion (0); xmlFree ((void *) 0);],
           [gl_cv_libxml=yes
            gl_cv_LIBXML="$LIBXML2 $LIBICONV"
            gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
@@ -80,10 +80,10 @@ AC_DEFUN([gl_LIBXML],
           fi
           if test -n "$libxml2_include_dir" && test -d "$libxml2_include_dir"; then
             CPPFLAGS="$gl_save_CPPFLAGS -I$libxml2_include_dir"
-            AC_TRY_LINK([[#include <libxml/xmlversion.h>
+            AC_TRY_LINK([#include <libxml/xmlversion.h>
                           #include <libxml/xmlmemory.h>
-                        ]],
-              [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                        ],
+              [xmlCheckVersion (0); xmlFree ((void *) 0);],
               [gl_cv_libxml=yes
                gl_cv_LIBXML="$LIBXML2 $LIBICONV"
                gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
--- gettext/gnulib-local/m4/libxml.m4
+++ gettext/gnulib-local/m4/libxml.m4
@@ -42,10 +42,10 @@ AC_DEFUN([gl_LIBXML],
       dnl -Wl,--disable-auto-import.
       AC_LIB_LINKFLAGS_BODY([xml2])
       LIBS="$gl_save_LIBS $LIBXML2 $LIBICONV"
-      AC_TRY_LINK([[#include <libxml/xmlversion.h>
+      AC_TRY_LINK([#include <libxml/xmlversion.h>
                     #include <libxml/xmlmemory.h>
-                  ]],
-        [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                  ],
+        [xmlCheckVersion (0); xmlFree ((void *) 0);],
         [gl_cv_libxml=yes
          gl_cv_LIBXML="$LIBXML2 $LIBICONV"
          gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
@@ -53,10 +53,10 @@ AC_DEFUN([gl_LIBXML],
       if test "$gl_cv_libxml" != yes; then
         gl_save_CPPFLAGS="$CPPFLAGS"
         CPPFLAGS="$CPPFLAGS $INCXML2"
-        AC_TRY_LINK([[#include <libxml/xmlversion.h>
+        AC_TRY_LINK([#include <libxml/xmlversion.h>
                       #include <libxml/xmlmemory.h>
-                    ]],
-          [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                    ],
+          [xmlCheckVersion (0); xmlFree ((void *) 0);],
           [gl_cv_libxml=yes
            gl_cv_LIBXML="$LIBXML2 $LIBICONV"
            gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
@@ -80,10 +80,10 @@ AC_DEFUN([gl_LIBXML],
           fi
           if test -n "$libxml2_include_dir" && test -d "$libxml2_include_dir"; then
             CPPFLAGS="$gl_save_CPPFLAGS -I$libxml2_include_dir"
-            AC_TRY_LINK([[#include <libxml/xmlversion.h>
+            AC_TRY_LINK([#include <libxml/xmlversion.h>
                           #include <libxml/xmlmemory.h>
-                        ]],
-              [[xmlCheckVersion (0); xmlFree ((void *) 0);]],
+                        ],
+              [xmlCheckVersion (0); xmlFree ((void *) 0);],
               [gl_cv_libxml=yes
                gl_cv_LIBXML="$LIBXML2 $LIBICONV"
                gl_cv_LTLIBXML="$LTLIBXML2 $LTLIBICONV"
