--- gettext-0.14.1/gettext-tools/misc/autopoint.in.orig	2004-01-29 22:17:27 +0300
+++ gettext-0.14.1/gettext-tools/misc/autopoint.in	2004-02-26 12:56:27 +0300
@@ -274,7 +274,8 @@ fi
 # users of CVS.
 #
 # Check availability of the CVS program.
-(cvs -v) >/dev/null 2>/dev/null || func_fatal_error "cvs program not found"
+CVS_EXE=/usr/bin/cvs
+("$CVS_EXE" -v) >/dev/null 2>/dev/null || func_fatal_error "$CVS_EXE: program not found"
 
 # Check in which directory config.rpath, mkinstalldirs etc. belong.
 auxdir=`cat "$configure_in" | grep '^AC_CONFIG_AUX_DIR' | sed -n -e 's/AC_CONFIG_AUX_DIR(\([^()]*\))/\1/p' | sed -e 's/^\[\(.*\)\]$/\1/' | sed -e 1q`
@@ -337,12 +338,13 @@ unset CVSWRAPPERS
 
 # Need to pass -d "$CVSROOT", because there may be a CVS directory in the
 # current directory.
-cvs -d "$CVSROOT" init
+"$CVS_EXE" -d "$CVSROOT" init
+subst 's/^LockDir/#LockDir/' "$CVSROOT/CVSROOT/config"
 gzip -d -c < "$gettext_dir/archive.tar.gz" | (cd "$cvs_dir" && tar xf -)
 
 cd "$work_dir"
 cvsver=gettext-`echo "$ver" | sed -e 's/\./_/g'`
-(cvs checkout -r"$cvsver" archive > /dev/null) 2>&1 | grep -v '^cvs checkout: Updating'
+("$CVS_EXE" checkout -r"$cvsver" archive > /dev/null) 2>&1 | grep -v '^cvs checkout: Updating'
 find archive -name CVS -type d -print | xargs rm -rf
 if test `find archive -type f -print | wc -l` = 0; then
   cd ..
