--- gettext/gettext-tools/misc/autopoint.in
+++ gettext/gettext-tools/misc/autopoint.in
@@ -456,7 +456,8 @@ case "@ARCHIVE_FORMAT@" in
     # (see <http://lists.gnu.org/archive/html/bug-cvs/2010-06/msg00011.html>).
     #
     # Check availability of the CVS program.
-    (cvs -v) >/dev/null 2>/dev/null || func_fatal_error "cvs program not found"
+    cvs=cvs
+    ($cvs -v) >/dev/null 2>/dev/null || func_fatal_error "cvs program not found"
 
     # Set up a temporary CVS repository.
     # We need the temporary CVS repository because any checkout needs write
@@ -501,12 +502,13 @@ case "@ARCHIVE_FORMAT@" in
 
     # Need to pass -d "$CVSROOT", because there may be a CVS directory in the
     # current directory.
-    cvs -d "$CVSROOT" init
+    $cvs -d "$CVSROOT" init
+    sed -i 's/^LockDir/#&/' "$CVSROOT/CVSROOT/config"
     gzip -d -c < "$gettext_dir/archive.cvs.tar.gz" | (cd "$cvs_dir" && tar xf -)
 
     cd "$work_dir"
     cvsver=gettext-`echo "$ver" | sed -e 's/\./_/g'`
-    (cvs -d "$CVSROOT" checkout -r"$cvsver" archive > /dev/null) 2>&1 | grep -v '^cvs checkout: Updating'
+    ($cvs -d "$CVSROOT" checkout -r"$cvsver" archive > /dev/null) 2>&1 | grep -v '^cvs checkout: Updating'
     find archive -name CVS -type d -print | xargs rm -rf
     cd ..
     rm -rf "$cvs_dir"
@@ -524,10 +526,11 @@ case "@ARCHIVE_FORMAT@" in
 
   git)
     # Check availability of the git program.
-    (git --version) >/dev/null 2>/dev/null || func_fatal_error "git program not found"
+    git=git
+    ($git --version) >/dev/null 2>/dev/null || func_fatal_error "git program not found"
     mkdir "$work_dir/archive"
     gzip -d -c < "$gettext_dir/archive.git.tar.gz" | (cd "$work_dir/archive" && tar xf -)
-    (cd "$work_dir/archive" && git checkout -q "gettext-$ver") || {
+    (cd "$work_dir/archive" && $git checkout -q "gettext-$ver") || {
       rm -rf "$work_dir"
       func_fatal_error "infrastructure files for version $ver not found; this is autopoint from GNU $package $version"
     }
