--- gettext/gettext-tools/misc/autopoint.in
+++ gettext/gettext-tools/misc/autopoint.in
@@ -429,6 +429,9 @@ if test `find archive -type f -print | wc -l` = 0; then
   rm -rf "$cvs_dir" "$work_dir"
   func_fatal_error "infrastructure files for version $ver not found; this is autopoint from GNU $package $version"
 fi
+find archive -type f |
+	xargs -r grep -l '\<sys_lib_\(dl\)\?search_path_spec=' |
+	xargs -r sed -i 's/\<sys_lib_\(dl\)\?search_path_spec=/#&/'
 cd ..
 
 # func_destfile file
