diff -uprk.orig gettext-0.14.1/gettext-tools/misc.orig/autopoint.in gettext-0.14.1/gettext-tools/misc/autopoint.in
--- gettext-0.14.1/gettext-tools/misc.orig/autopoint.in	2004-09-13 18:24:25 +0400
+++ gettext-0.14.1/gettext-tools/misc/autopoint.in	2004-09-13 18:46:55 +0400
@@ -33,52 +33,13 @@ gettext_dir="@datadir@/gettext"
 # Support for relocatability.
 func_find_curr_installdir ()
 {
-  # Determine curr_installdir, even taking into account symlinks.
-  curr_executable="$0"
-  case "$curr_executable" in
-    */* | *\\*) ;;
-    *) # Need to look in the PATH.
-      if test "${PATH_SEPARATOR+set}" != set; then
-        { echo "#! /bin/sh"; echo "exit 0"; } > /tmp/conf$$.sh
-        chmod +x /tmp/conf$$.sh
-        if (PATH="/nonexistent;/tmp"; conf$$.sh) >/dev/null 2>&1; then
-          PATH_SEPARATOR=';'
-        else
-          PATH_SEPARATOR=:
-        fi
-        rm -f /tmp/conf$$.sh
-      fi
-      save_IFS="$IFS"; IFS="$PATH_SEPARATOR"
-      for dir in $PATH; do
-        IFS="$save_IFS"
-        test -z "$dir" && dir=.
-        for exec_ext in ''; do
-          if test -f "$dir/$curr_executable$exec_ext"; then
-            curr_executable="$dir/$curr_executable$exec_ext"
-            break 2
-          fi
-        done
-      done
-      IFS="$save_IFS"
-      ;;
-  esac
-  # Make absolute.
-  case "$curr_executable" in
-    /* | ?:/* | ?:\\*) ;;
-    *) curr_executable=`pwd`/"$curr_executable" ;;
-  esac
-  # Resolve symlinks.
-  while : ; do
-    lsline=`LC_ALL=C ls -l "$curr_executable"`
-    case "$lsline" in
-      *" -> "*)
-        curr_executable=`echo "$curr_executable" | sed -e 's,/[^/]*$,,'`/`echo "$lsline" | sed -n -e 's,^.* -> \(.*\),\1,p'` ;;
-      *) break ;;
-    esac
-  done
-  curr_installdir=`echo "$curr_executable" | sed -e 's,/[^/]*$,,'`
-  # Canonicalize.
-  curr_installdir=`cd "$curr_installdir" && pwd`
+	[ -n "$0" ] || return 1
+	curr_executable="$(type -p "$0")" || return 1
+	[ "$curr_executable" != "${curr_executable##/}" ] || return 1
+	curr_executable="$(readlink -ev "$curr_executable")" || return 1
+	[ "$curr_executable" != "${curr_executable##/}" ] || return 1
+	curr_installdir="${curr_executable%/*}"
+	[ "$curr_installdir" != "${curr_installdir##/}" ] || return 1
 }
 func_find_prefixes ()
 {
@@ -103,7 +64,7 @@ if test "@RELOCATABLE@" = yes; then
   exec_prefix="@exec_prefix@"
   bindir="@bindir@"
   orig_installdir="$bindir" # see Makefile.am's *_SCRIPTS variables
-  func_find_curr_installdir # determine curr_installdir
+  func_find_curr_installdir || func_fatal_error func_find_curr_installdir # determine curr_installdir
   func_find_prefixes
   # Relocate the directory variables that we use.
   gettext_dir=`echo "$gettext_dir/" | sed -e "s%^${orig_installprefix}/%${curr_installprefix}/%" | sed -e 's,/$,,'`
diff -uprk.orig gettext-0.14.1/gettext-tools/misc.orig/gettextize.in gettext-0.14.1/gettext-tools/misc/gettextize.in
--- gettext-0.14.1/gettext-tools/misc.orig/gettextize.in	2004-09-13 18:24:25 +0400
+++ gettext-0.14.1/gettext-tools/misc/gettextize.in	2004-09-13 18:49:37 +0400
@@ -33,52 +33,13 @@ gettext_dir="@datadir@/gettext"
 # Support for relocatability.
 func_find_curr_installdir ()
 {
-  # Determine curr_installdir, even taking into account symlinks.
-  curr_executable="$0"
-  case "$curr_executable" in
-    */* | *\\*) ;;
-    *) # Need to look in the PATH.
-      if test "${PATH_SEPARATOR+set}" != set; then
-        { echo "#! /bin/sh"; echo "exit 0"; } > /tmp/conf$$.sh
-        chmod +x /tmp/conf$$.sh
-        if (PATH="/nonexistent;/tmp"; conf$$.sh) >/dev/null 2>&1; then
-          PATH_SEPARATOR=';'
-        else
-          PATH_SEPARATOR=:
-        fi
-        rm -f /tmp/conf$$.sh
-      fi
-      save_IFS="$IFS"; IFS="$PATH_SEPARATOR"
-      for dir in $PATH; do
-        IFS="$save_IFS"
-        test -z "$dir" && dir=.
-        for exec_ext in ''; do
-          if test -f "$dir/$curr_executable$exec_ext"; then
-            curr_executable="$dir/$curr_executable$exec_ext"
-            break 2
-          fi
-        done
-      done
-      IFS="$save_IFS"
-      ;;
-  esac
-  # Make absolute.
-  case "$curr_executable" in
-    /* | ?:/* | ?:\\*) ;;
-    *) curr_executable=`pwd`/"$curr_executable" ;;
-  esac
-  # Resolve symlinks.
-  while : ; do
-    lsline=`LC_ALL=C ls -l "$curr_executable"`
-    case "$lsline" in
-      *" -> "*)
-        curr_executable=`echo "$curr_executable" | sed -e 's,/[^/]*$,,'`/`echo "$lsline" | sed -n -e 's,^.* -> \(.*\),\1,p'` ;;
-      *) break ;;
-    esac
-  done
-  curr_installdir=`echo "$curr_executable" | sed -e 's,/[^/]*$,,'`
-  # Canonicalize.
-  curr_installdir=`cd "$curr_installdir" && pwd`
+	[ -n "$0" ] || return 1
+	curr_executable="$(type -p "$0")" || return 1
+	[ "$curr_executable" != "${curr_executable##/}" ] || return 1
+	curr_executable="$(readlink -ev "$curr_executable")" || return 1
+	[ "$curr_executable" != "${curr_executable##/}" ] || return 1
+	curr_installdir="${curr_executable%/*}"
+	[ "$curr_installdir" != "${curr_installdir##/}" ] || return 1
 }
 func_find_prefixes ()
 {
@@ -103,7 +64,7 @@ if test "@RELOCATABLE@" = yes; then
   exec_prefix="@exec_prefix@"
   bindir="@bindir@"
   orig_installdir="$bindir" # see Makefile.am's *_SCRIPTS variables
-  func_find_curr_installdir # determine curr_installdir
+  func_find_curr_installdir || func_fatal_error func_find_curr_installdir # determine curr_installdir
   func_find_prefixes
   # Relocate the directory variables that we use.
   gettext_dir=`echo "$gettext_dir/" | sed -e "s%^${orig_installprefix}/%${curr_installprefix}/%" | sed -e 's,/$,,'`
